geode-core/src/distributedTest/java/org/apache/geode/internal/cache/ClientServerTransactionDistributedTest.java

```diff
@@ -0,0 +1,152 @@
+import org.apache.geode.test.dunit.rules.ClientCacheRule;
+import org.apache.geode.test.dunit.rules.DistributedRule;
+import org.apache.geode.test.junit.rules.serializable.SerializableTestName;
+
+@SuppressWarnings("serial")
+public class ClientServerTransactionDistributedTest implements Serializable {
+
+  private String hostName;
+  private String uniqueName;
+  private String regionName;
+  private VM server1;
+  private int port1;
+
+  @Rule
+  public DistributedRule distributedRule = new DistributedRule();
+
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+
+  @Rule
+  public ClientCacheRule clientCacheRule = new ClientCacheRule();
+
+  @Rule
+  public SerializableTestName testName = new SerializableTestName();
+
+  @Before
+  public void setUp() {
+    server1 = getVM(0);
+    hostName = getHostName();
+    uniqueName = getClass().getSimpleName() + "_" + testName.getMethodName();
+    regionName = uniqueName + "_region";
+  }
+
+  @Test
+  public void clientTransactionIDAboveIntegerMaxValue() {
+    port1 = server1.invoke(() -> createServerRegion());
+
+    // Test that transaction ID overflow to one
+    TXManagerImpl.INITIAL_UNIQUE_ID_VALUE = Integer.MAX_VALUE;
+    createClientRegion(true, port1);
+
+    TXManagerImpl txManager =
+        (TXManagerImpl) clientCacheRule.getClientCache().getCacheTransactionManager();
+
+    txManager.begin();
+    TXStateProxyImpl txStateProxy = (TXStateProxyImpl) txManager.getTXState();
+    int transactionID = ((TXId) txStateProxy.getTransactionId()).getUniqId();
+
+    int numOfOperations = 5;
+    putData(numOfOperations);
+    txManager.commit();
+
+    server1.invoke(() -> verifyTransactionResult(numOfOperations));
+    assertEquals(1, transactionID);
+  }
+
+  private void putData(int numberOfEntries) {
+    Region region = clientCacheRule.getClientCache().getRegion(regionName);
+    for (int key = 0; key < numberOfEntries; key++) {
+      String value = getValue(key);
+      region.put(key, value);
+    }
+  }
+
+  private void verifyTransactionResult(int numberOfEntries) {
+    Region region = cacheRule.getCache().getRegion(regionName);
+    for (int i = 0; i < numberOfEntries; i++) {
+      LogService.getLogger().info("region get key {} value {} ", i, region.get(i));
+    }
+    for (int i = 0; i < numberOfEntries; i++) {
+      assertEquals("value" + i, region.get(i));
+    }
+  }
+
+  private int createServerRegion() throws Exception {
+    PartitionAttributesFactory factory = new PartitionAttributesFactory();
+    PartitionAttributes partitionAttributes = factory.create();
+    cacheRule.getOrCreateCache().createRegionFactory(RegionShortcut.PARTITION)
+        .setPartitionAttributes(partitionAttributes).create(regionName);
+    CacheServer server = cacheRule.getCache().addCacheServer();
+    server.setPort(0);
+    server.start();
+    return server.getPort();
+  }
+
+  private void createClientRegion(boolean connectToFirstPort, int... ports) {
```

Comment:
This method is only used once, so `connectToFirstPort` is always true and `ports.length` is always equal to 1. Could it be simplified to remove unused code?

Reply:
done